backup/backup.py

```diff
@@ -192,7 +199,8 @@ def read_metadata(self):
     def add_change(self, entry: Change) -> bool:
         typ = b'\x01' if entry.snapshot is None else b'\x02'
         if typ == b'\x01':
-            payload = b'\x00'.join([t.encode('UTF-8') for t in entry.transaction])
+            stmts = [t.encode('UTF-8') if isinstance(t, str) else t for t in entry.transaction]
+            payload = b'\x00'.join(stmts)
```

Comment:
FWIW, in #189 I have resolved this the other way around. Change always contains `str`'s, and they are encoded and decoded as bytes only where needed. I found this slightly easier to handle (require less conversions in total, and `_rewrite_stmt` can use straightforward `re`).

Reply:
Good point, I'm fine with either, since this is at runtime only and doesn't change the backup files.

Feel free to switch it around if that makes it easier to rebase.

Sorry for breaking your PR by merging this. I'll review #189 asap.

Reply:
Thanks—yeah no problem, these changes are very contained and easy to port over.